src/main/java/io/wagz/statements/domain/Transaction.java

```diff
@@ -0,0 +1,20 @@
+package io.wagz.statements.domain;
+
+import io.wagz.statements.constants.TransactionType;
+import java.math.BigDecimal;
+
+public record Transaction(String description, BigDecimal amount, TransactionType transactionType) {
+  public static Transaction ofSignedAmount(BigDecimal amount, String description) {
+    if (amount.compareTo(BigDecimal.ZERO)
+        > 0) { // positive amount is credit and negative amount is debit
+      return new Transaction(description, amount.abs(), TransactionType.CREDIT);
+    } else {
```

Comment:
we could use a ternary operatior here `Transaction transaction = amount.compareTo(BigDecimal.ZERO) > 0 ? new Transaction(description, amount.abs(), TransactionType.CREDIT): Transaction(description, amount.abs(), TransactionType.DEBIT); 
return transaction;`